lib/cib/cib_utils.c

```diff
@@ -485,13 +485,26 @@ cib_perform_op(const char *op, int call_options, cib__op_fn_t fn, bool is_query,
 
             /* Does the CIB support the "update-*" attributes... */
             if (current_schema >= minimum_schema) {
+                /* Ensure values of origin, client, and user in scratch match
+                 * the values in req
+                 */
                 const char *origin = crm_element_value(req, F_ORIG);
+                const char *client = crm_element_value(req, F_CIB_CLIENTNAME);
 
                 CRM_LOG_ASSERT(origin != NULL);
-                crm_xml_replace(scratch, XML_ATTR_UPDATE_ORIG, origin);
-                crm_xml_replace(scratch, XML_ATTR_UPDATE_CLIENT,
-                                crm_element_value(req, F_CIB_CLIENTNAME));
-                crm_xml_replace(scratch, XML_ATTR_UPDATE_USER, crm_element_value(req, F_CIB_USER));
+                crm_xml_add(scratch, XML_ATTR_UPDATE_ORIG, origin);
```

Comment:
Since `CRM_LOG_ASSERT()` doesn't keep us from proceeding, we should still NULL-check `origin` to decide whether to add or remove it. I'll make a note for a future PR.

Reply:
Yep, and we can drop the assert -- it was probably only useful when this code was first being developed, and nothing horrible happens if it's NULL